### PR TITLE
docs: add rspack-start

### PR DESCRIPTION
**Pull request by 9aoy**
## Types of changes

- [x] Docs change / Dependency upgrade
- [ ] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/builder/builder-doc/docs/zh/guide/rspack-start.mdx [hunk 1]
```diff
@@ -0,0 +1,158 @@
+# 使用 Rspack
+
+## 启用 Rspack
+
+### 在 Modern.js 框架中启用
+
+Modern.js 框架默认使用 Webpack Builder 作为构建引擎。如果希望启用 Rspack Builder 作为构建引擎，我们推荐通过 [Modern.js 生成器](https://modernjs.dev/apis/app/commands/new.html#%E5%90%AF%E7%94%A8%E5%8F%AF%E9%80%89%E5%8A%9F%E8%83%BD) 方式启用。
```
Comment (chenjiahan) [inline packages/builder/builder-doc/docs/zh/guide/rspack-start.mdx:7]: 对外不要透出 Webpack Builder 这个名词，只有 Webpack Provider 和 Rspack Provider

File: packages/builder/builder-doc/docs/zh/guide/rspack-start.mdx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+1. 安装 Builder（builder 版本号需要与 modern.js 版本号保持一致）。
+
+```
+pnpm add @modern-js/builder-rspack-provider@2.4.0 -D
+```
+
+通过 package.json 或 lock 文件，确认 modern.js 版本号:
+
+```
```
Comment (chenjiahan) [inline packages/builder/builder-doc/docs/zh/guide/rspack-start.mdx:19]: \`\`\`json

File: packages/builder/builder-doc/docs/zh/guide/rspack-start.mdx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+});
+```
+
+
+## 从 webpack 迁移到 Rspack
+
+Builder 旨在消除不同打包工具之间的主要差异，帮助用户以较低成本进行迁移。但由于各个打包工具的能力不同，部分差异仍需用户了解。
+
+### 配置差异
+目前，Builder 内超过 65% 的配置项已经适配了 Rspack。Builder 所做的工作包括不限于：
```
Comment (chenjiahan) [inline packages/builder/builder-doc/docs/zh/guide/rspack-start.mdx:63]: 缺少空行，记得 prettier 一下

File: packages/builder/builder-doc/docs/zh/guide/rspack-start.mdx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  - [output.disableInlineRuntimeChunk](/api/config-output.html#outputdisableinlineruntimechunk)
+  - [output.enableInlineScripts](/api/config-output.html#outputenableinlinescripts)
+  - [output.enableInlineStyles](/api/config-output.html#outputenableinlinestyles)
+- tsChecker 能力
+  - [output.disableTsChecker](/api/config-output.html#outputdisabletschecker)
+
+#### Experiments Config
+> Builder 中的一些实验性配置，实验性配置可以开启 Builder 中尚未稳定的功能。
+
+所有 experiments 下的配置项及能力不支持在 rspack 内使用。
```
Comment (chenjiahan) [inline packages/builder/builder-doc/docs/zh/guide/rspack-start.mdx:108]: rspack -> Rspack

File: packages/builder/builder-doc/docs/zh/guide/rspack-start.mdx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  - [tools.webpackChain](/api/config-tools.html#toolswebpackchain)
+
+
+:::tip
+除上述配置外，一些已支持的配置可能存在能力差异，请参考各配置 API 获取具体差异信息。
+:::
+
+
+### 修改 Rspack 配置
+你可以使用 tools.rspack 来修改 rspack 配置对象，对 rspack 配置进行更灵活的操作。
```
Comment (chenjiahan) [inline packages/builder/builder-doc/docs/zh/guide/rspack-start.mdx:143]: tools.rspack 后面补个文档

Comment (9aoy) [reply packages/builder/builder-doc/docs/zh/guide/rspack-start.mdx]: next PR

File: packages/builder/builder-doc/docs/zh/guide/advanced/rspack-start.mdx [hunk 6]
```diff
@@ -0,0 +1,173 @@
+# 使用 Rspack
+
+[Rspack](https://www.rspack.org/) 是字节跳动 Web Infra 团队自研的 Rust Bundler，核心架构对齐 webpack 实现。目前 Rspack Provider 已适配并验证了 Builder 内超过 65% 的配置能力。
```
Comment (chenjiahan) [inline packages/builder/builder-doc/docs/zh/guide/advanced/rspack-start.mdx:3]: 「超过 65% 」改成「大部分」吧

File: packages/builder/builder-doc/docs/zh/guide/advanced/rspack-start.mdx [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+1. 安装 Rspack Provider（版本号需要与 modern.js 版本号保持一致）。
+
+```
+pnpm add @modern-js/builder-rspack-provider@2.4.0 -D
+```
+
+通过 package.json 或 lock 文件，确认 modern.js 版本号:
+
+```json
+// package.json
```
Comment (chenjiahan) [inline packages/builder/builder-doc/docs/zh/guide/advanced/rspack-start.mdx:24]: json 里不能有注释，可以改成 jsonc 试试

另外 json 上下的括号可以写完整

File: packages/builder/builder-doc/docs/zh/guide/advanced/rspack-start.mdx [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+  "devDependencies": {
+    "@modern-js/app-tools": "2.4.0",
+    "@modern-js/eslint-config": "2.4.0",
+```
+
+2. 声明使用
+
+```
```
Comment (chenjiahan) [inline packages/builder/builder-doc/docs/zh/guide/advanced/rspack-start.mdx:32]: 少了 ts

File: packages/builder/builder-doc/docs/zh/guide/advanced/rspack-start.mdx [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+// modern.config.ts
```
Comment (chenjiahan) [inline packages/builder/builder-doc/docs/zh/guide/advanced/rspack-start.mdx:33]: 文件名不用写成注释，直接写 \`\`\`ts title=modern.config.ts 更美观

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**